llama-index-integrations/readers/llama-index-readers-web/llama_index/readers/web/knowledge_base/base.py

```diff
@@ -125,9 +127,10 @@ def scrape_article(
         return {"title": title, "subtitle": subtitle, "body": body, "url": url}
 
     def get_article_urls(
-        self, browser: Any, root_url: str, current_url: str
+        self, browser: Any, root_url: str, current_url: str, max_depth: int = 100
```

Comment:
shouldn't we track the url we visited so we do not visit it again, instead of hardcoding depth ?

Reply:
I needed a quick fix but yeah, that would be a better solution, feel free to open a PR!